### ISSUE TITLE
New AI behaviors: Team/Squad AI

### ISSUE THREAD

**Issue by jose21crisis** [eezstreet/SWATEliteForce#211]
From a Ghost Recon Forum: "Ghost Recon's AI works in hierarchies, starting at the highest level with a "Company AI", which basically just tells NPC the side they're fighting on and who the enemies are, so they know whether or not to attack another character they encounter. Beneath the company level, there are "Platoon AI", which allow basic communication and coordination between subordinate AI units (mainly managing alertness, awareness of player presence, location of player detection). Finally, at the bottom of the AI hierarchy are the "Team AI", each one controlling a team of up to six NPC, assigning the team's overall strategy and coordinating the actions of its individual members, allowing complex team efforts, e.g. bounding overwatch with suppressive fire in support of offensive "fire and maneuver" forward pushing or defensive "center peel" disengagement."

Basically I want to add the "Team AI" from this text. It would appear only with high skilled suspects. It will allow these suspects to employ new tactics, like flanking, fire and maneuver, suppressive fire, among others. It will allow these suspects to work together as a team instead of the standard Individual AI that's used in vanilla SWAT 4/TSS/SEF v5

**Comment by eezstreet** [comment]
From what I understand, all you really need is a new suspect Hive type.
